packages/core/logging/core-logging-server-internal/src/appenders/rolling_file/policies/time_interval/get_next_rolling_time.ts

```diff
@@ -23,8 +23,10 @@ export const getNextRollingTime = (
     const increment =
       interval.get(incrementedUnit) -
       (currentMoment.get(incrementedUnit) % interval.get(incrementedUnit));
-    const incrementInMs = moment.duration(increment, incrementedUnit).asMilliseconds();
-    return currentMoment.startOf(incrementedUnit).toDate().getTime() + incrementInMs;
+    const nextRollingMoment = currentMoment
+      .startOf(incrementedUnit)
+      .add(increment, incrementedUnit);
+    return nextRollingMoment.toDate().getTime();
```

Comment:
Not sure why I performed the increment manually here... Turns out, moment does it better by correctly taking DST switch into account...